### ISSUE TITLE
client: Accept-Encoding header

### ISSUE THREAD

**Issue by stokito** [valyala/fasthttp#1397]
My client sends requests and expects gzip in the response. Now I have to the the AcceptEncoding as a regular header:

	req.Header.Set(fasthttp.HeaderAcceptEncoding, "deflate,gzip")

But internally it will allocate the Header argsKV storage.
How about to store the AcceptEncoding as a field similar to others essential headers like contentType, contentLengthBytes, host and userAgent?
Will you accept a PR for this?

As a downside this will make the struct memory footprint bigger. But it also can make the server's HasAcceptEncodingBytes() faster because it need to check it each time.

I also thinking about the optimization of the argsKV.
Basically there is not so many of standard headers that are in active use: Authorization, CT, CL, CE, Accept-Range, UA, Host. It's up to 10 of them in one typical request.
Maybe we can use some kind of 8-bit bitmask instead of []byte key array. This will allow to make faster scans.

**Comment by erikdubbelboer** [comment]
Sounds good but might be complicated. Keep in mind that an Accept-Encoding header can look like this for example: `Accept-Encoding: deflate, gzip;q=0.9, *;q=0.5`. Not sure how you want to store that as something non-allocating in the struct.

**Comment by stokito** [comment]
Are these priorities just the same as an order? I mean 'deflate;q=0.5, gzip;q=0.9' is just the same as resorted 'gzip, deflate'

**Comment by erikdubbelboer** [comment]
I'm not sure, you'll have to read the RFC I guess.
